src/share/import.js

```diff
@@ -0,0 +1,57 @@
+import fs from '../lib/fs';
+import Buffer from '../lib/buffer';
+import Path from '../lib/path';
+import WebTorrent from 'webtorrent';
+const sh = new fs.Shell();
+
+document.getElementById('btnDownload').addEventListener('click', startDownload);
+console.log('inside of import!');
+
+const downloadAllTorrentFiles = torrent =>
+    Promise.all(
+        torrent.files.map(
+            file =>
+                new Promise((resolve, reject) => {
+                    var filename = Path.join('/', file.path);
+                    file.getBuffer((err, buffer) => {
+                        if (err) {
+                            return reject(err);
+                        }
+                        fs.writeFile(filename, new Buffer(buffer), err => {
+                            if (err) {
+                                console.error(
+                                    'Error writing file, error was ',
+                                    err,
+                                    filename
+                                );
+                                return reject(err);
+                            } else {
+                                console.log('Wrote to ', filename);
+                                resolve();
+                            }
+                        });
+                    });
+                })
+        )
+    );
+
+function startDownload() {
+    let magnetURI = document.getElementById('torrentId').value;

```

Comment:
We should probably also look in the URL (`window.location.search`) to see if we have something like `?magnet...` and use that if so (i.e., pre-populate your form element).

Reply:
The page will now look at the URL for a magnet URI and insert that into the text field.

`document.getElementById('torrentId').value = window.location.search.substring(1);`